### DISCUSSION TITLE
Good project structure for front & backend

### DISCUSSION THREAD

**Discussion by MarkoPavi** [community/community#32069] [New to GitHub] [answered]
Hey I got a question. 
Say I have a backend project and a corresponding frontend project that communicates via REST. How do I make them both part of the same git repository or 2 repositories part of a git project. I already tried that but its just an issue tracker.
Is there a possibilty of a sub-repository? I just want them to be seperate when I commit/push but organized in the same folder neatly. 
Sorry if its the wrong category any mods can freely move this accoringly.

**Comment by averagebit** [comment, marked as answer]
I believe what you're looking for are git [submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules).

In short if you run `git submodule add <your-repo>` in a git repository, `<your-repo>` will be added as a folder inside your main repository. You can add as many submodules as you like. You could even have private submodules in public repositories and they would not be visible to the public. An example of that could be seen in my [dotfiles](https://github.com/averagebit/.dot) repository with the `personal` submodule.

Other than that you could also create a [monorepo](https://monorepo.tools/) essentially combining both projects under the same repository which is common practice in certain situations as well. There are [build systems](https://nx.dev/) dedicated to managing such repositories.

**Reply by MarkoPavi** [reply]
Thanks . I tested it and it works.

**Comment by Helu99** [comment]
Can someone give me a video of how to create a frontend and backend by using VS Code and GitHub please.
